pgx-tests/src/framework.rs

```diff
@@ -162,9 +204,12 @@ fn initialize_test_framework(
     postgresql_conf: Vec<&'static str>,
 ) -> eyre::Result<(LogLines, String)> {
     let mut state = TEST_MUTEX.lock().unwrap_or_else(|_| {
-        // if we can't get the lock, that means it was poisoned,
-        // so we just abruptly exit, which cuts down on test failure spam
-        std::process::exit(1);
+        // This used to immediately throw an std::process::exit(1), but it
+        // would consume both stdout and stderr, resulting in error messages
+        // not being displayed unless you were running tests with --nocapture.
+        panic!(
+            "Could not obtain test mutex. Please check for errors in pervious tests to find the root cause."
+        );
```

Comment:
I do believe this was the reason none of the errors were showing up at all. `std::process::exit(1)` would completely consume any errors that wanted to be displayed in the output. I am not 100% sure `panic!` is the right way to go, but in my local testing the failure "spam" wasn't all that significant. I felt like it was better to have more information than not enough.

Reply:
This implies something about test history without stating the cause outright. How about
```suggestion
        // This used to immediately throw an std::process::exit(1), but it
        // would consume both stdout and stderr, resulting in error messages
        // not being displayed unless you were running tests with --nocapture.
        panic!("Could not obtain test mutex. A previous test may have hard-aborted while holding it.");
```

Reply:
Looks good! We should use that.